### PR TITLE
Profile Items

### PR DESCRIPTION
**Pull request by ErdemT09**
We are now able to display items specific to a profile. Both for user's own profile and others' profiles.
image
image

The code structure used here needs some fixes, so I will be improving it in the next PRs.

### CODE REVIEW

File: src/components/items/ItemList.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-import { fetchItems } from '../../redux/actions/item-list-action.js';
+import { fetchItems, fetchItemsByProfile, fetchItemsByProfileUsername} from '../../redux/actions/item-list-action.js';
 import { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
 import TagPanel from '../TagPanel.js';
 
 const PAGE_ITEMS_COUNT = 9;
 
-const ItemList = ({ fetchItems, items, addToCart, totalItemCount, isInProfile }) => {
+const ItemList = ({ fetchItems, items, addToCart, totalItemCount, isInProfile, fetchItemsByProfile, profileId, fetchItemsByProfileUsername, isInUserProfile }) => {
```
Comment (ErdemT09) [inline src/components/items/ItemList.js:11]: Adding more props for each use case is definitely something to be fixed.

File: src/redux/actions/item-list-action.js [hunk 2]
```diff
@@ -6,36 +6,56 @@ export const fetchItems = (pageNumber, tag) => {
     
     if(tag !== undefined)
     {
-       return  fetchProductsByTag(pageNumber, tag, dispatchType)
```
Comment (ErdemT09) [inline src/redux/actions/item-list-action.js:9]: For this entire file, we can maybe find a more modular or a generalized function structure.

File: src/components/items/ItemList.js (hunk 1 above)
Comment (altayhunoglu) [reply src/components/items/ItemList.js:11]: we can just use "props".

**Review by altayhunoglu — APPROVED** [review]
Thank you for the efforts, as always Erdem.

File: src/components/items/ItemList.js (hunk 1 above)
Comment (ErdemT09) [reply src/components/items/ItemList.js:11]: Although, we wouldn't have parameters nicely listed with it, it's really better than having 50 variables listed out.

**Pull request merged**